### ISSUE TITLE
Error message in FTV settings - PHP 7.1 problem

### ISSUE THREAD

**Issue by mpwt** [JustCarmen/fancy_treeview#53]
Surprisingly, I found an error message in the FTV settings few days ago:

![bildschirmfoto 2017-02-19 um 12 03 28](https://cloud.githubusercontent.com/assets/16311627/23101768/d70d2274-f69b-11e6-982f-235bea0c48b5.png)

Fancy Treeview works on my site but FTV PDF not. At the moment I use webtrees 1.7.9 with all master aditions from 19.01.2017 and the recent master of FTV and FTV PDF 1.7.9.1.

Do you have any idea?

**Comment by mpwt** [comment]
This must be a PHP 7.1 problem. Just I have tested it with PHP 7.0 and there isn't the error message from above and FTV PDF works again ... I was wondering about that all works as I have updated on 12.01. and not at the moment and I remembered that I changed the PHP version few days ago.

**Comment by JustCarmen** [comment]
This is a mpdf problem. It is reported earlier at their issue tracker.

According to this message it should have been fixed: https://github.com/mpdf/mpdf/issues/278
But another related issue is still open: https://github.com/mpdf/mpdf/issues/163

As you can see Php 7.1 compatibility is still in development state so I guess you can't rely on it right now.

**Comment by mpwt** [comment]
After uninstalling Fancy Treeview PDF and deleting the whole FTV PDF folder from modules_V3 the error message is shown yet with PHP 7.1. I'm not sure that mpdf and PHP 7.1 is the problem.

**Comment by mpwt** [comment]
It seems that the following function in FancyTreeviewClass.php have a problem with PHP 7.1:

Line 217-236
```
	/**
	 * Get a list of all used countries
	 *
	 * @return list
	 */
	protected function getCountryList() {
		$list	 = '';
		$sql	 = "SELECT SQL_CACHE p_place as country FROM `##places` WHERE p_parent_id=:parent_id AND p_file=:tree_id";
		$args	 = array(
			'parent_id'	 => '0',
			'tree_id'	 => $this->tree()->getTreeId()
		);

		$countries = Database::prepare($sql)->execute($args)->fetchAll(PDO::FETCH_ASSOC);

		foreach ($countries as $country) {
			$list[$country['country']] = $country['country']; // set the country as key to display as option value.
		}
		return $list;
	}
```

**Comment by JustCarmen** [comment]
I don't have Php 7.1 yet, so I have to install it first and then I'll take a look.

**Comment by mpwt** [comment]
Hi Carmen, could you check the error with PHP 7.1?

**Comment by JustCarmen** [comment]
Hi I am sorry for the late response. But I've already fixed this issue a while ago with commit 9fae44fae5c9882f64a03d01d35f288076e0672b but forgot to mark this issue as solved.
